source/common/http/header_map_impl.cc

```diff
@@ -483,7 +483,7 @@ void HeaderMapImpl::clear() {
   cached_byte_size_ = 0;
 }
 
-size_t HeaderMapImpl::removeIf(const std::function<bool(const HeaderEntry&)>& predicate) {
+size_t HeaderMapImpl::removeIf(const HeaderMap::HeaderMatchPredicate& predicate) {
```

Comment:
FWIW I think you can drop the HeaderMap qual here as HeaderMapImpl is a prefix.

Reply:
It's not actually (it's a mixin).

Reply:
got it; makes sense.